### PR TITLE
Linalg to XSMM: AddOp conversion

### PR DESCRIPTION
**Pull request by chelini**

### CODE REVIEW

File: lib/TPP/ConvertLinalgToXsmm.cpp [hunk 1]
```diff
@@ -260,6 +301,101 @@ struct ConvertGenericToUnaryRelu : public OpRewritePattern<linalg::GenericOp> {
...
+        genericOp.getMatchingIndexingMap(rhsOperand), /*operandIdx=*/1);
+    if (failed(broadCastFlagRhs))
+      return failure();
+
+    auto flagLhs =
+        xsmm::BinaryFlagsAttr::get(rewriter.getContext(), *broadCastFlagLhs);
+    auto flagRhs =
+        xsmm::BinaryFlagsAttr::get(rewriter.getContext(), *broadCastFlagRhs);
+
+    // Spaghetti code to handle 'NONE' as it conflicts with other flags; we
```
Comment (adam-smnk) [inline lib/TPP/ConvertLinalgToXsmm.cpp:379]: Where does this conflict come from?

File: test/Conversion/LinalgToXsmm/linalg-to-binary.mlir
Comment (adam-smnk) [inline test/Conversion/LinalgToXsmm/linalg-to-binary.mlir:1]: Please add a test case with no broadcast

File: lib/TPP/ConvertLinalgToXsmm.cpp [hunk 2]
```diff
@@ -260,6 +301,101 @@ struct ConvertGenericToUnaryRelu : public OpRewritePattern<linalg::GenericOp> {
...
+    // best solution is to get rid of it.
+    ArrayAttr flags;
+    if (flagLhs.getValue() != xsmm::BinaryFlags::NONE &&
+        flagRhs.getValue() != xsmm::BinaryFlags::NONE) {
+      flags = rewriter.getArrayAttr({flagLhs, flagRhs});
+    } else if (flagLhs.getValue() != xsmm::BinaryFlags::NONE) {
+      flags = rewriter.getArrayAttr({flagLhs});
+    } else if (flagRhs.getValue() != xsmm::BinaryFlags::NONE) {
+      flags = rewriter.getArrayAttr({flagRhs});
+    }
```
Comment (adam-smnk) [inline lib/TPP/ConvertLinalgToXsmm.cpp:390]: Doesn't this need one more default case to add NONE?

File: lib/TPP/ConvertLinalgToXsmm.cpp (hunk 1 above)
Comment (chelini) [reply lib/TPP/ConvertLinalgToXsmm.cpp:379]: the verifier. If we have "NONE" in the flags no other flags is allowed.

File: lib/TPP/ConvertLinalgToXsmm.cpp (hunk 2 above)
Comment (chelini) [reply lib/TPP/ConvertLinalgToXsmm.cpp:393]: yes..

**Pull request merged**